### PR TITLE
Changes for correct work with ransack

### PR DESCRIPTION
**Pull request by one-more-alex**
This changes were needed to make Ajax search work with Ransack and Active admin

### CODE REVIEW

File: app/inputs/filter_ajax_filter_input.rb [hunk 1]
```diff
@@ -32,13 +31,16 @@ def input_html_options
...
     if association.options && association.options.keys.include?(:class_name)
-      filter_class = association.options[:class_name].constantize rescue @object.klass
+      filter_class = association.options[:class_name].constantize rescue ransack.klass
     else
-      filter_class = association_name.classify.constantize rescue @object.klass
+      filter_class = association_name.classify.constantize rescue ransack.klass
     end
-    selected_value = @object.conditions.find { |c| c.attributes.map(&:name).include?(method.to_s) }.values.first.value rescue nil
+    selected_value = ransack.conditions.find {|c|
+        c.attributes.map(&:name).include?(method.to_s)
```
Comment (monkeyci) [inline app/inputs/filter_ajax_filter_input.rb]: Use 2 (not 4) spaces for indentation.

File: app/inputs/filter_ajax_filter_input.rb [hunk 2]
```diff
@@ -32,13 +31,16 @@ def input_html_options
...
+    association = ransack.klass.reflect_on_association(association_name.to_sym)
     if association.options && association.options.keys.include?(:class_name)
-      filter_class = association.options[:class_name].constantize rescue @object.klass
+      filter_class = association.options[:class_name].constantize rescue ransack.klass
     else
-      filter_class = association_name.classify.constantize rescue @object.klass
+      filter_class = association_name.classify.constantize rescue ransack.klass
     end
-    selected_value = @object.conditions.find { |c| c.attributes.map(&:name).include?(method.to_s) }.values.first.value rescue nil
+    selected_value = ransack.conditions.find {|c|
```
Comment (monkeyci) [inline app/inputs/filter_ajax_filter_input.rb]: Avoid using {...} for multi-line blocks.<br>Space between { and | missing.

File: app/inputs/filter_ajax_filter_input.rb [hunk 3]
```diff
@@ -32,13 +31,15 @@ def input_html_options
...
     if association.options && association.options.keys.include?(:class_name)
-      filter_class = association.options[:class_name].constantize rescue @object.klass
+      filter_class = association.options[:class_name].constantize rescue ransack.klass
     else
-      filter_class = association_name.classify.constantize rescue @object.klass
+      filter_class = association_name.classify.constantize rescue ransack.klass
     end
-    selected_value = @object.conditions.find { |c| c.attributes.map(&:name).include?(method.to_s) }.values.first.value rescue nil
+    selected_value = ransack.conditions
+      .find { |c| c.attributes.map(&:name).include?(method.to_s) }.values.first.value rescue nil
```
Comment (monkeyci) [inline app/inputs/filter_ajax_filter_input.rb]: Align .find with .conditions on line 41.

**Comment by ldlsegovia** [comment]
This was fixed in the middle of a big refactor https://github.com/platanus/activeadmin_addons/blob/master/lib/activeadmin_addons/support/input_helpers/filter_input_methods.rb#L9
thanks anyway

**Pull request closed without merging**